### DISCUSSION TITLE
Possible operability with CLIFORCE_COLOR?

### DISCUSSION THREAD

**Discussion by mjpieters** [donatj/force-color.org#97] [Ideas]
There was a pre-existing effort to standardise an environment variable for CLI tools to force colors: https://bixense.com/clicolors/ / https://github.com/jhasse/clicolors.

How does this project differ from the pre-existing effort, and should there be interoperability with that pre-existing standardisation?

**Comment by donatj** [comment]
Hmm, I was not even aware of the priors existence. 

It doesn't seem to be very widely cited or supported? 

https://www.google.com/search?q=link%3Ahttps%3A%2F%2Fbixense.com%2Fclicolors%2F

The big difference is really just that this is just trying to standardize the `FORCE_COLOR` variable which was already pretty widely used across the node ecosystem, as largely standardized by node itself.

Given how little used the previous standard was, I'm not sure interoperability would be worth the effort? Feel free to push back on that.

**Comment by mjpieters** [comment]
I discovered it because I noticed that the [Rust `anstyle` library](https://docs.rs/anstyle/latest/anstyle/) supports the `CLICOLORS` / `CLICOLORS_FORCE` / `NO_COLOR` environment variables already. It is fairly popular with [142 known dependent projects](https://crates.io/crates/anstyle/reverse_dependencies), including `cargo`, the rust environment manager. Asking projects that already use `anstyle` to support `FORCE_COLOR` either push back to say they already support that standard or implement `FORCE_COLOR` as an alias for `CLICOLORS_FORCE`. 

I think that at least setting out guidelines for how the two standards can interact would be helpful.

**Comment by mjpieters** [comment]
On popularity, google searches for [`CLICOLOR`](https://www.google.com/search?q=CLICOLOR) and [`CLICOLOR_FORCE`](https://www.google.com/search?q=CLICOLOR_FORCE) produce a much larger number of results.

**Comment by Andrej730** [comment]
Agree that https://force-color.org/ in theory should at least mention https://bixense.com/clicolors/ (and vice versa) to direct users as a possible workaround where `FORCE_COLOR` is not supported. I've found `FORCE_COLOR` page first, then went investigating how to make it work for cmake, searched cmake docs and only then found existence of `CLICOLOR_FORCE`.

**Comment by baodrate** [comment]
FWIW, did a little research and the most significant uses of `CLICOLOR_FORCE` I could find (+minimum version supported):

* FreeBSD's `ls`: v3.5 (2000-08-12) (26 years ago, wow):
  * [freebsd#20483](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=20483)
  * [freebsd@3d2ddc9](https://cgit.freebsd.org/src/commit/?id=3d2ddc9e1a6a60d585b7c0a3e378b6ef986dac57) (or on github: freebsd/freebsd-src@3d2ddc9e1a6a60d585b7c0a3e378b6ef986dac57)
  * No doubt from where macOS got this behavior
* cmake: v3.5 (2015-10-16)
  * [cmake#15789](https://cmake.org/Bug/view.php?id=15789)
  * [cmake@9c6a7203](https://gitlab.kitware.com/cmake/cmake/-/commit/9c6a7203) (or on github: Kitware/CMake@9c6a7203fb0c6034771011a8ac58ae447726b919)
* tree: v1.8.0 (2018-11-16)
  * Old-Man-Programmer/tree@4031b9ea9dccc9cff7a5b30e329214d89a53d8f6

Certainly `FORCE_COLOR` has had more traction than `CLICOLOR_FORCE`, but the significance of the BSD usage via macOS, seems significant enough to warrant consideration

Also relevant:

* jhasse/clicolors#15

**Comment by mjpieters** [comment]
It seems that the clicolors project has [decided to direct people to this standard as the way forward for future projects](https://github.com/jhasse/clicolors/issues/15#issuecomment-4121206956), which is very commendable.

**Comment by BatmanAoD** [comment]
I suggest simply adding something like this to the FAQ section:

> Is this compatible with other standars for color-control variables, such as https://bixense.com/clicolors/?

Yes, as long as `FORCE_COLOR` has the highest precedence. For instance, when using https://bixense.com/clicolors/, it should be treated as an alias for `CLICOLOR_FORCE`, so that setting either or both of `FORCE_COLOR` and `CLICOLOR_FORCE` always enables color.

**Reply by BatmanAoD** [reply]
...actually, I misunderstood the precedence. According to https://bixense.com/clicolors/, `FORCE_COLOR` + `NO_COLOR` should result in no color: so the precedence goes `NO_COLOR` > `CLICOLOR_FORCE` = `FORCE_COLOR` > `CLICOLOR`.
